### ISSUE TITLE
Hi team,  We're experiencing a build issue on Android:  react-native:0.66.3 react-native-helpscout-beacon:0.4.0 com.helpscout:beacon-ui:2.3.1 Build log error  Manifest merger failed : Attribute meta-data#com.google.firebase.messaging.default_notification_channel_id@value value=() from [:react-native-firebase_messaging] AndroidManifest.xml:37:13-29 is also present at [com.helpscout:beacon-ui:2.3.1] AndroidManifest.xml:69:13-70 value=(@string/hs_beacon_notification_channel_id). Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:35:9-37:32 to override. Any ideas?

### ISSUE THREAD

**Issue by patrick-samy** [helpscout/beacon-android-sdk-sample#178]
Hi team,

We're experiencing a build issue on Android after adding the Helpscout SDK to our project.

**Technical details**

- Beacon version: 2.3.1
- Android OS version: 30
- Device make/model: N/A

**Actual outcome**

Build log:
```
Manifest merger failed : Attribute meta-data#com.google.firebase.messaging.default_notification_channel_id@value value=() from [:react-native-firebase_messaging] AndroidManifest.xml:37:13-29
is also present at [com.helpscout:beacon-ui:2.3.1] AndroidManifest.xml:69:13-70 value=(@string/hs_beacon_notification_channel_id).
Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:35:9-37:32 to override.
```

**Expected outcome**

The Android app should build successfully after adding the Helpscout Android SDK.

**Steps to reproduce**

- Using `com.helpscout:beacon-ui:2.3.1` via `react-native-helpscout-beacon:0.4.0`
- Add the Android SDK `com.helpscout:beacon-ui` to `build.gradle`
- Run `./gradlew assembleRelease`

**Comment by adrianne-helpscout** [comment]
@patrick-samy We do not build or support the react-native-helpscout package. I'm afraid you will have to take up your issues with that library with the developers. You might also mention to the the developers of that package that we have a much newer version of the SDK available. Our current version is 3.0.2 and will have an even newer version arriving very shortly. Version 2.3.1 is over a year old now.

**Comment by patrick-samy** [comment]
Thanks @adrianne-helpscout. I've updated the dependency locally to use the latest version of the beacon SDK for Android.

Unfortunately, it seems like the same manifest conflict is happening with `com.helpscout:beacon:4.0.0`:
```
> Task :app:processReleaseMainManifest FAILED
[com.helpscout:beacon:4.0.0] /Users/patrick/.gradle/caches/transforms-3/6e4838f7c602a14141ad634fff530559/transformed/jetified-beacon-4.0.0/AndroidManifest.xml Warning:
        Package name 'com.helpscout.beacon.ui' used in: com.helpscout:beacon:4.0.0, com.helpscout:beacon-ui:2.3.2.
[io.insert-koin:koin-android:3.1.3] /Users/patrick/.gradle/caches/transforms-3/f14e73376975e2864b0167b3845065f9/transformed/jetified-koin-android-3.1.3/AndroidManifest.xml Warning:
        Package name 'org.koin.android' used in: io.insert-koin:koin-android:3.1.3, org.koin:koin-android:2.2.1.
/Users/patrick/Projects/span/app/android/app/src/main/AndroidManifest.xml:68:13-70 Error:
        Attribute meta-data#com.google.firebase.messaging.default_notification_channel_id@value value=(@string/hs_beacon_notification_channel_id) from [com.helpscout:beacon:4.0.0] AndroidManifest.xml:68:13-70
        is also present at [:react-native-firebase_messaging] AndroidManifest.xml:37:13-29 value=().
        Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:66:9-68:73 to override.

See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.

ReactNativeFirebase WARNING: NPM package '@react-native-firebase/app-check' depends on '@react-native-firebase/app' v13.1.0 but found v12.6.1, this might cause build issues or runtime crashes.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processReleaseMainManifest'.
> Manifest merger failed with multiple errors, see logs

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 46s
382 actionable tasks: 7 executed, 375 up-to-date
```

**Comment by adrianne-helpscout** [comment]
@patrick-samy I'm curious if you are still including react-native-helpscout-beacon? 
Warning:
        Package name 'com.helpscout.beacon.ui' used in: com.helpscout:beacon:4.0.0, com.helpscout:beacon-ui:2.3.2.
        
helpscout.beacon.ui use to be a library that we renamed [removed in version 3.0.0](https://github.com/helpscout/beacon-android-sdk-sample/blob/main/CHANGELOG.md#version-300-2020-03-30)

When we released 3.0.0 it was a breaking change that required changes to your build.gradle file. Details are in the linked changelog above.

**Comment by aliripe** [comment]
Hey @patrick-samy! Alicia here joining Adrianne! 👋 

As Adrianne correctly mentioned, version 2.3.2 is very outdated. You should contact the developers or fork the library and upgrade that dependency to 4.0.0 to get the latest bug fixes and enhancements.

Updating it locally, that is, trying to force the Reactive Native library to use version 4.0.0, won't work because of the issue mentioned above, there were some breaking changes and we merged `beacon-core` and `beacon-ui` into a single module called `beacon`, so Gradle sees 2.3.2 and 4.0.0 as completely different dependencies and that's creating conflicts.

However, even if you bump the library to 4.0.0 – which is highly recommended – you're likely still going to run into the original issue you reported:
```
Attribute meta-data#com.google.firebase.messaging.default_notification_channel_id@value value=(@string/hs_beacon_notification_channel_id) from [com.helpscout:beacon:4.0.0] AndroidManifest.xml:68:13-70
is also present at [:react-native-firebase_messaging] AndroidManifest.xml:37:13-29 value=().
Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:66:9-68:73 to override.
```

And you should apply the suggestion in the error message to circunvent this:
> Suggestion: add `tools:replace="android:value"` to `<meta-data>` element at `AndroidManifest.xml:66:9-68:73` to override.

Which translates to:
```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:tools="http://schemas.android.com/tools"> <!-- Add namespace declaration -->

  …

  <meta-data
      android:name="com.google.firebase.messaging.default_notification_channel_id"
      android:value="@string/<your_string_resource>"
      tools:replace="android:value"/> 

</manifest>
```

I hope this helps!
